src/GraphQL.Tests/Subscription/ShouldlyExtensions.cs

```diff
@@ -57,4 +58,16 @@ public static ExecutionResult ShouldBeSuccessful(this ExecutionResult result)
         result.Executed.ShouldBeTrue();
         return result;
     }
+
+    public static void ShouldBeWhen(this object? obj, bool condition, object onTrue, object onFalse) =>
+        obj.ShouldBe(condition ? onTrue : onFalse);
```

Comment:
Why?

```cs
obj.ShouldBeWhen(condition, trueObj, falseObj);
// vs
obj.ShouldBe(condition ? trueObj : falseObj);
```
Seems the second option is at least as preferable as the first, especially since `ShouldBeWhen` isn't a generic method.  It's at least as readable, perhaps more, and it's shorter.